source/module_hamilt_lcao/module_gint/grid_technique.h

```diff
@@ -1,126 +1,226 @@
-	std::vector<gridIntegral::gridIndex> nnrg_index;
-    
     // public functions
-	public:
-
-	Grid_Technique();
-	~Grid_Technique();
-
-	void set_pbc_grid(
-			const int &ncx_in,
-			const int &ncy_in,
-			const int &ncz_in,
-			const int &bx_in,
-			const int &by_in,
-			const int &bz_in,
-			const int &nbx_in,
-			const int &nby_in,
-			const int &nbz_in,
-			const int &nbxx_in,
-			const int &nbzp_start_in,
-            const int& nbzp_in,
-            const int& ny,
-            const int& nplane,
-            const int& startz_current);
+  public:
+    Grid_Technique();
+    ~Grid_Technique();
+
+    void set_pbc_grid(const int& ncx_in,
+                      const int& ncy_in,
+                      const int& ncz_in,
+                      const int& bx_in,
+                      const int& by_in,
+                      const int& bz_in,
+                      const int& nbx_in,
+                      const int& nby_in,
+                      const int& nbz_in,
+                      const int& nbxx_in,
+                      const int& nbzp_start_in,
+                      const int& nbzp_in,
+                      const int& ny,
+                      const int& nplane,
+                      const int& startz_current);
 
     /// number of elements(basis-pairs) in this processon
     /// on all adjacent atoms-pairs(Grid division)
     void cal_nnrg(Parallel_Orbitals* pv);
-    int cal_RindexAtom(const int& u1, const int& u2, const int& u3, const int& iat2) const;
-    
-private:
+    int cal_RindexAtom(const int& u1,
+                       const int& u2,
+                       const int& u3,
+                       const int& iat2) const;
 
+  private:
     void cal_max_box_index(void);
-    
+
     int maxB1;
     int maxB2;
-	int maxB3;
+    int maxB3;
 
-	int minB1;
-	int minB2;
-	int minB3;
+    int minB1;
+    int minB2;
+    int minB3;
 
-	int nB1;
-	int nB2;
-	int nB3;
+    int nB1;
+    int nB2;
+    int nB3;
 
     int nbox;
 
-	// atoms on meshball
-	void init_atoms_on_grid(const int& ny, const int& nplane, const int& startz_current);
-	void init_atoms_on_grid2(const int* index2normal);
-	void cal_grid_integration_index(void);
-	void cal_trace_lo(void);
-	void check_bigcell(int* &ind_bigcell, bool* &bigcell_on_processor);
-	void get_startind(const int& ny, const int& nplane, const int& startz_current);
+    // atoms on meshball
+    void init_atoms_on_grid(const int& ny,
+                            const int& nplane,
+                            const int& startz_current);
+    void init_atoms_on_grid2(const int* index2normal);
+    void cal_grid_integration_index(void);
+    void cal_trace_lo(void);
+    void check_bigcell(int*& ind_bigcell, bool*& bigcell_on_processor);
+    void get_startind(const int& ny,
+                      const int& nplane,
+                      const int& startz_current);
+
+#if ((defined __CUDA) /* || (defined __ROCM) */)
+  public:
+    double* ylmcoef_g;
```

Comment:
initialize all pointers with "nullptr", here write the code as " double* ylmcoef_g=nullptr;"